pkg/controller/license/trial/trial_controller.go

```diff
@@ -179,11 +179,16 @@ func (r *ReconcileTrials) startTrialActivation() error {
 
 func (r *ReconcileTrials) completeTrialActivation(license types.NamespacedName) (reconcile.Result, error) {
 	if r.trialState.CompleteTrialActivation() {
-		status, err := licensing.ExpectedTrialStatus(r.operatorNamespace, license, r.trialState)
+		expectedStatus, err := licensing.ExpectedTrialStatus(r.operatorNamespace, license, r.trialState)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-		return reconcile.Result{}, r.Update(&status)
+		var existingStatus corev1.Secret
+		if err := r.Get(k8s.ExtractNamespacedName(&expectedStatus), &existingStatus); err != nil {
```

Comment:
This is mainly for the overeager fake client which requires a resource version on every update. The fact that we are hitting this issue only now tells me that none of the other tests executed this code path 🤷

Reply:
How about calling `ReconcileSecret()` instead? Manipulating the resource version here feels a bit weird.